server/src/test/java/org/elasticsearch/index/seqno/RetentionLeaseIT.java

```diff
@@ -95,11 +97,21 @@ public void testRetentionLeasesSyncedOnAdd() throws Exception {
             final CountDownLatch latch = new CountDownLatch(1);
             final ActionListener<ReplicationResponse> listener = ActionListener.wrap(r -> latch.countDown(), e -> fail(e.toString()));
             // simulate a peer recovery which locks the soft deletes policy on the primary
-            final Closeable retentionLock = randomBoolean() ? primary.acquireRetentionLock() : () -> {};
+            final Closeable retentionLock = randomBoolean() ? primary.acquireHistoryRetentionLock(Engine.HistorySource.INDEX) : () -> {};
             currentRetentionLeases.put(id, primary.addRetentionLease(id, retainingSequenceNumber, source, listener));
             latch.await();
             retentionLock.close();
+        }
 
+        for (int i = 0; i < length; i++) {
+            final String id = randomFrom(currentRetentionLeases.keySet());
+            final CountDownLatch latch = new CountDownLatch(1);
+            primary.removeRetentionLease(id, countDownLatchListener(latch));
+            // simulate a peer recovery which locks the soft deletes policy on the primary
+            final Closeable retentionLock = randomBoolean() ? primary.acquireHistoryRetentionLock(Engine.HistorySource.INDEX) : () -> {};
+            currentRetentionLeases.remove(id);
+            latch.await();
+            retentionLock.close();
```

Comment:
why is that needed, where is this coming from?

Reply:
It is part of the `testRetentionLeasesSyncedOnAdd` at the state of the backport https://github.com/crate/crate/blob/c54f405b817d51d97ba1b621e1f421ca9efccda7/server/src/test/java/org/elasticsearch/index/seqno/RetentionLeaseIT.java#L106 and it was introduced here https://github.com/elastic/elasticsearch/pull/37398.

Reply:
Afaik this code block is part of a missing `testRetentionLeaseSyncedOnRemove` test and should not be added inside the `testRetentionLeasesSyncedOnAdd` test method. See  https://github.com/elastic/elasticsearch/blob/b9fbc8dc74826329f0bb00fc4f531507a12ec327/server/src/test/java/org/elasticsearch/index/seqno/RetentionLeaseIT.java

Reply:
oh, you are right, my fault. thank you!

Reply:
I took the wrong diff.